### PR TITLE
Updated per-container proxy support

### PR DESCRIPTION
**Pull request by groovecoder**

### CODE REVIEW

**Comment by groovecoder** [comment]
@maxxcrawford or @bakulf - if either of you have time, I think this can be reviewed now. It builds on the previous PR which only supported `http` proxy types to include support for `https`, `socks`, and `socks4` proxies. Some things to consider:

* Should we store the proxy for the container in the same object as the the site assigments? That way it will be included when containers are sync'd?
* Should we remove [the "bogus proxy" approach to blocking requests without a `tabId`](https://github.com/mozilla/multi-account-containers/pull/1284/files#r310342224) ? And change it to use `onBeforeRequest` handler?
* Something else we should consider?

**Comment by kendallcorner** [comment]
FYI, I don't believe you will simply be able to add it to the same object as the site assignments to get it to sync.

**Comment by Chronophylos** [comment]
@groovecoder are you still working on this?

**Comment by groovecoder** [comment]
Updated this to resolve conflicts. I'm going to see if I can finish this up soon.

**Comment by groovecoder** [comment]
Updated to resolve conflicts and fix eslint. @Chronophylos were you interested in contributing to this?

**Comment by Chronophylos** [comment]
No, I just want the feature :)

**Comment by groovecoder** [comment]
Heh, yeah it's a very popular request. I'm checking around to see if I can get some last considerations ... considered.

File: src/manifest.json [hunk 1]
```diff
@@ -27,7 +27,8 @@
     "storage",
     "tabs",
     "webRequestBlocking",
-    "webRequest"
+    "webRequest",
+    "proxy"
```
Comment (kendallcorner) [inline src/manifest.json:31]: If you are adding the proxy permission, which is an [Advised Permission](https://extensionworkshop.com/documentation/develop/request-the-right-permissions/#advised-permissions), you might also want to add 'bookmarks' and remove the [approval request](https://github.com/mozilla/multi-account-containers/blob/4b56a2f0bb91cd9a12b95121fcff0c8e96ff66bd/src/js/options.js#L6) in `options.js`

**Review by kendallcorner — COMMENTED** [review]
Hi!

**Comment by nobodysu** [comment]
Thank you for your work. Too bad it aren't merged yet.
I have to ask, does it address favicon loading, either by blocking them or channeling through specific proxy? If not, the user could be fingerprinted by opening the per-container site list in extension settings.

**Review by maxxcrawford — APPROVED** [review]
Works as expected! I added a few nit/one-liner CSS changes. I think we can land this! 🛩️

File: src/manifest.json (hunk 1 above)
Comment (maxxcrawford) [reply src/manifest.json:32]: Filed this as a seperate issue: https://github.com/mozilla/multi-account-containers/issues/2138

**Pull request merged**